docs/src/views/ThemingExamples.tsx

```diff
@@ -0,0 +1,400 @@
+        `      color: 'darkred',`,
+        `      typeSecondaryColor: '#ffffff',`,
+        `      typeSecondaryBorderColor: 'transparent',`,
+        `      typeSecondaryBackgroundColor: '#6699CC',`,
+        `      typeSecondaryBackgroundColorHover: '#91A3B0',`,
+        `    },`,
+        `  },`,
+        `  componentStyles: {`,
+        `    Button: {`,
+        `      icon: {`,
+        `        fontSize: '12px',`,
+        `      },`,
+        `    },`,
+        `  },`,
+        `}`,
+        ``,
+        `const provider = () => (`,
+        `  <Provider theme={theme}>`,
+        `    <div>`,
+        `      <Button content="Button" />`,
+        `      <Button type="primary" icon="plus" iconOnly />`,
+        `      <Button type="secondary" icon="at" content="Send email" />`,
+        `      <Icon name="chess rook" size="big" />`,
+        `      <Label content="Label with icon" icon="close" />`,
+        `      <br /><br />`,
+        `      <Divider type="primary">Branding primary</Divider>`,
+        `      <Divider type="secondary">Branding secondary</Divider>`,
+        `    </div>`,
+        `  </Provider>`,
+        `)`,
+        ``,
+        `export default provider`,
+      ].join('\n')}
+      render={() => (
+        <div>
+          <Provider
+            theme={{
+              siteVariables: {
+                brand: 'darkred',
+                brand04: '#8F5873',
+                gray08: '#A8516E8C',
+                gray06: '#f4c2c2',
+                gray03: '#757575',
+              },
+              componentVariables: {
+                Button: {
+                  height: '24px',
+                  minWidth: '24px',
+                  borderRadius: '8px',
+                  color: 'darkred',
+                  typeSecondaryColor: '#ffffff',
+                  typeSecondaryBorderColor: 'transparent',
+                  typeSecondaryBackgroundColor: '#6699CC',
+                  typeSecondaryBackgroundColorHover: '#91A3B0',
+                },
+              },
+              componentStyles: {
+                Button: {
+                  icon: {
+                    fontSize: '12px',
+                  },
+                },
+              },
+            }}
+          >
+            <div>
+              <Button content="Button" />
+              <Button type="primary" icon="plus" iconOnly />
+              <Button type="secondary" icon="at" content="Send email" />
+              <Icon name="chess rook" size="big" />
+              <Label content="Label with icon" icon="close" />
+              <br />
+              <br />
+              <Divider content="Primary Divider" type="primary" />
+              <Divider content="Secondary Divider" type="secondary" />
+            </div>
+          </Provider>
+        </div>
+      )}
+    />
+
+    <Header as="h3" content="Nesting Providers" />
+    <p>
+      If you need to customize a bigger part of the theme, you can nest the second provider and
```

Comment:
probably, it would be good to mentioning common use cases where one will need to preserver most of theming aspects, but at the same time achieve one or more of the following
- tweak some specific styling aspects of existing theme - so to speak, build theme on top of existing one
- introduce theme changes that will be applied to specific components' subtree only

This will immediately provide reader with an idea whether this section is something she is looking for